src/socks-common/socks-headers.ts

```diff
@@ -176,7 +176,11 @@ export function composeAuthHandshakeBuffer(auths:Auth[]) : ArrayBuffer {
   var handshakeBytes = new Uint8Array(auths.length + 2);
   handshakeBytes[0] = VERSION5;
   handshakeBytes[1] = auths.length;
-  handshakeBytes.set(auths, 2);
+  // TODO: Revert to set() when Uint8Array's type definition is fixed in TS1.6:
```

Comment:
I think it would be better to use a type-cast here, as you don't have to have different compiled JS code, e.g. : 
```
(<any>handshakeBytes).set(auths, 2);
```
Or you might explore if you can create a local over-ride at the top of the file. 
```
interface Uint8Array {
  set(auths:Auth[], n:number);
}
```
Same for other uses of set.